walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,142 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long requestDuration = request.getDuration();
+    Collection<String> requestAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    ArrayList<TimeRange> alreadyBookedTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> alreadyBookedTimesWOptional = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> noOverlap = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> noOverlapWOptional = new ArrayList<TimeRange>();    
+    ArrayList<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> availableTimesWOptional = new ArrayList<TimeRange>();  
+    int newStart;
+    int newEnd;
```

Comment:
nit: Could we declare these closer to their usage? I think it'll help improve readability a bit.

Reply:
Yes, addressed in "address nits(clarify greeter and its testing, revise variable declaration, fix indentation)"